assets/javascripts/discourse/widgets/qa-comment.js

```diff
@@ -8,24 +8,86 @@ import { ajax } from "discourse/lib/ajax";
 import bootbox from "bootbox";
 import I18n from "I18n";
 
+createWidget("qa-comment-editor", {
+  tagName: "div",
+  buildKey: (attrs) => `qa-comment-editor-${attrs.id}`,
+
+  buildClasses(attrs) {
+    return ["qa-comment-editor", `qa-comment-editor-${attrs.id}`];
+  },
+
+  defaultState(attrs) {
+    return { updatingComment: false, value: attrs.raw };
+  },
+
+  html(attrs, state) {
+    return [
+      h("textarea", attrs.raw),
+      this.attach("button", {
+        action: "editComment",
+        disabled: state.updatingComment,
+        contents: I18n.t("qa.post.edit_comment"),
+        icon: "pencil-alt",
+        className: "btn-primary qa-comment-editor-submit",
+      }),
+      this.attach("link", {
+        action: "collapseEditor",
+        className: "qa-comment-editor-cancel",
+        contents: () => I18n.t("qa.post.cancel_comment"),
+      }),
+    ];
+  },
+
+  input(e) {
+    this.state.value = e.target.value;
+  },
+
+  keyDown(e) {
+    if (e.key === "Enter" && (e.ctrlKey || e.metaKey)) {
```

Comment:
I think this is a nice user shortcut. Do we usually log them somewhere like in a readme on how to use it? Not sure if plugins have that similar thing in core where they show users a set of shortcuts.

Reply:
`ctrl + enter` is quite a common keyboard shortcut in forms so I don't think we need to document this. For now at least, I don't think plugins that introduce special shortcuts have a way to extend the [keyboard shortcuts helper](https://github.com/discourse/discourse/blob/main/app/assets/javascripts/discourse/app/lib/keyboard-shortcuts.js) in core.